core/includes/image.inc

```diff
@@ -526,6 +550,65 @@ function image_hex2rgba(&$hex) {
   );
 }
 
+/**
+ * Get an SVG image's defined dimensions.
+ *
+ * @param string $uri
+ *   The URI or path to an SVG image.
+ *
+ * @return array|FALSE
+ *   An array containing the keys "width" and "height" as integer values. If the
+ *   image is an SVG but no set dimensions are available, these keys will have
+ *   NULL values. If the image is not an SVG, or the SVG file is empty or
+ *   invalid, FALSE will be returned.
+ */
+function image_get_svg_dimensions($uri) {
+  // Safety check.
+  if (!image_is_svg($uri)) {
+    return FALSE;
+  }
+
+  // Return cached dimensions if already retrieved once this request.
+  $cached_images = &backdrop_static(__FUNCTION__, array());
+  if (isset($cached_images[$uri])) {
+    return $cached_images[$uri];
+  }
+
+  $svg_dimensions = array(
+    'width' => IMAGE_SVG_DEFAULT_WIDTH,
+    'height' => IMAGE_SVG_DEFAULT_HEIGHT,
+  );
+
+  $file_contents = file_get_contents($uri);
+  if ($file_contents && $svg = simplexml_load_string($file_contents)) {
+    foreach ($svg->attributes() as $attribute => $value) {
+      if ($attribute === 'width' || $attribute === 'height') {
+        $svg_dimensions[$attribute] = (int) $value;
+      }
+    }
+  }
+  else {
+    return FALSE;
```

Comment:
The return of FALSE doesn't provide value here, I belief.
We could simply stick with defaults in this case. Or am I misunderstanding something?

Reply:
My thinking is that without explicitly returning FALSE here, for an invalid or empty file, the function would return the default dimensions, which wouldnt make sense really.

Reply:
Makes sense...
This utility function could get called on arbitrary URIs or actually strings.